covid_files.R

```diff
@@ -0,0 +1,205 @@
+head(df3)
+colnames(df3)[2] <- 'sc2_amplicons';#
+clade
+head(summary1)
+head(clade)
+summary1<-merge(summary1,df2, by.x="df1", by.y= "taxon", all.x=TRUE)
+summary1<-merge(summary1,df3, by.x="df1", by.y= "Sample", all.x=TRUE)
+summary1<-merge(summary1,clade, by.x="df1", by.y= "seqName", all.x=TRUE)
+
+collectionD<-select(ngs_file,sampleNumber,collectionDate)
+summary1<-merge(summary1,collectionD, by.x="df1", by.y= "sampleNumber", all.x=TRUE)
+
+
+# Reading fasta files for nucleotides count: actg & anbiguous (n)
+# ========================
+num_actgx<-vector();num_nx<-vector(); pass_fail<-vector()
+runPath1 = paste(runPath,"covidseq_output/Sample_Analysis/",sep ="/")
+num_actgx = 0; num_nx = 0;
+n
+head(summary1)
+for (i in 1:n){
+  sample=summary1$df1[i]; sample; 
+  amplicons=summary1$sc2_amplicons[i];amplicons
+  if(amplicons >=50)
+  {
+    Lyapunovv<- Sys.glob(paste(runPath1,sample,"*/*.fasta",sep = ""),dirmark = TRUE)
+    Lyapunovv
+    a =  str_count(read.fasta(Lyapunovv, as.string = T, forceDNAtolower = T), pattern = "a|c|t|g");num_actgx[i] = a
+    b =  str_count(read.fasta(Lyapunovv, as.string = T, forceDNAtolower = T))-a; num_nx[i] = b; pass_fail[i] ="pass"; c = "pass"
+   
+  }else {# AmpliconThereshold <50,
+    a = b =0; num_actgx[i] = 0; num_nx[i] = 0; pass_fail[i] ="fail"; c = "fail"
+  }
+  cat(sample," \t",a,"\t",b,"\t",c,"\t",i,"\n")
+}
+  
+  df_new <- as.data.frame(cbind(as.character(num_actgx), as.character(num_nx),pass_fail)); 
+  df_new<-df_new %>% rename(num_actg = 1, num_n = 2)
+  head(df_new)
+  summary<-cbind(summary1,df_new);
+  head(summary)
+  
+  # Adding more  info to the summary file
+  # ==============================
+  summary<-merge(summary,df4, by.x = "df1", by.y = "SampleID", all.x = T )
+  head(summary)
+  colnames(summary)[1]<-c("SampleNumber")
+  summary<-cbind(summary,'Run Name'=args[1]) 
+ 
+  # Saving file summary file at  /Volumes/NGS/Analysis/covidseq/$runName
+  # ==========================
+  d=paste(runPath,'Rsumcovidseq',sep="/");dir.create(file.path(d,"" ), recursive = TRUE); p1<-Sys.time()
+  dx<- paste(d,'/Rcovidseq_summary-',paste0(Sys.Date(),'-',hour(Sys.time()),'-',minute(Sys.time()),'-',second(Sys.time())),".csv",sep=""); 
+  write.csv(summary,dx,row.names=FALSE)
+  cat(crayon::bold$underline("Summary is complete and can be found at: ",d,"\n"))
+  head(summary)
+  
+  # Creating a file for NGS tracking system for copy/paste
+  # ===========================
+  dx<- paste(d,'/NGS_Tracking-',paste0(Sys.Date(),'-',hour(Sys.time()),'-',minute(Sys.time()),'-',second(Sys.time())),".csv",sep=""); 
+  write.csv(select(summary,"SampleNumber","Sample_ID","collectionDate","Run Name","lineage","num_actg","clade")
+            ,dx,row.names=FALSE)
+  
+  # json file: fastas2json  
+  # ========================
+  path2python<-paste("python3 /home/Bioinformatics/Dripping_Rock/bin/fasta_to_json.py",args[1],sep = " ")
+  path2python
+  system(path2python)
```

Comment:
Is there any plan to replace this with something else?

Reply:
I have a version in R that can be added to this script but the version in python is working fine.